apis/core/v1alpha2/core_types.go

```diff
@@ -324,3 +336,26 @@ type ApplicationConfigurationList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []ApplicationConfiguration `json:"items"`
 }
+
+// DataOutput specifies a data output source from an object.
+type DataOutput struct {
+	// Name is the unique name of a DataOutput in an ApplicationConfiguration.
+	Name string `json:"name,omitempty"`
+
+	// FieldPath refers to the value of an object's field.
+	FieldPath string `json:"fieldPath,omitempty"`
+}
+
+// DataInput specifies a data input sink to an object.
+type DataInput struct {
+	// ValueFrom specifies the value source.
+	ValueFrom DataInputValueFrom `json:"valueFrom,omitempty"`
+
+	// ToFieldPaths specifies the field paths of an object to fill passed value.
+	ToFieldPaths []string `json:"toFieldPaths,omitempty"`
```

Comment:
How about renaming `ToFieldPaths` to `FieldPath` which is same to that of `DataOutput`? 1) This seems consistent. 2) `To` in `ToFieldPaths` sounds like `Out` in `DataOutput` which might lead to confusion as I did right now.